website/addons/s3/static/tests/s3NodeConfig.test.js

```diff
@@ -80,16 +80,90 @@ var s3ViewModelSettings = {
 
 describe('s3NodeConfigViewModel', () => {
     describe('isValidBucketName', () => {
-        assert.isTrue(isValidBucketName('valid'));
-        assert.isFalse(isValidBucketName('not.valid', false));
-        assert.isFalse(isValidBucketName('no'));
-        assert.isFalse(isValidBucketName(''));
-        var chars = [];
-        for (var i = 0, len = 64; i < len; i++) {
-            chars.push('a');
-        }
-        var tooLong = chars.join('');
-        assert.isFalse(isValidBucketName(tooLong));
+    var chars = [];
+    for (var i = 0; i < 63; i++) {
+        chars.push('a');
+    }
+        var longName = chars.join('');
+
+    var moreChars = [];
+    for (var j = 0; j < 255; j++) {
+        moreChars.push('a');
+    }
+    var reallyLongName = moreChars.join('');
+
+        it('allows these names in strict mode', (done) => {
+            assert.isTrue(isValidBucketName('valid'), 'basic label');
+            assert.isTrue(isValidBucketName('also.valid'), 'two labels');
+            assert.isTrue(isValidBucketName('pork.beef.chicken'), 'three labels');
+            assert.isTrue(isValidBucketName('a-o.valid'), 'label w/ hyphen');
+            assert.isTrue(isValidBucketName('11.12.m'), 'numbers as labels');
+            assert.isTrue(isValidBucketName('aa.22.cc'), 'mixed label types');
+            assert.isTrue(isValidBucketName('a------a'), 'multiple hypens');
+            assert.isTrue(isValidBucketName(longName), 'name up to 63 characters');
+            done();
+        });
+
+        it('DOES NOT allow these names in strict mode', (done) => {
```

Comment:
Minor: using `done` is only necessary for async tests. For sync tests, you can omit it.